### PR TITLE
LPS-126295

### PR DESCRIPTION
**Pull request by alexcandidos**

### CODE REVIEW

**Comment by Tim-Cao** [comment]
Just start reviewing

**Comment by Tim-Cao** [comment]
Hi @alexcandidos, this case is duplicate with FieldGroups#DuplicateFieldGroupOnStructure

**Comment by alexcandidos** [comment]
hey @Tim-Cao , are you referring to the empty stub `FieldGroups#DuplicateFieldGroupOnStructure` rt? If yes, it is okay, discussing with @emanuelfisilva , we believe that this one was wrongly added to the FieldGroups.testcase file. 


So, DuplicateAFieldGroup should be enough and we will not automate `FieldGroups#DuplicateFieldGroupOnStructure`.

**Comment by Tim-Cao** [comment]
Hi @alexcandidos, DuplicateFieldGroupOnStructure is already implemented.

**Comment by emanuelfisilva** [comment]
Hey @Tim-Cao , my latest version of master don't have this test implemented. Would you kindly point out the PR were this test is implemented?

**Comment by Tim-Cao** [comment]
Sorry, my mistake. I regard https://github.com/liferay/liferay-portal/blob/master/modules/apps/data-engine/data-engine-test/src/testFunctional/tests/wem/structures/FieldGroups.testcase#L103 as DuplicateFieldGroupOnStructure
I will review this one today.

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/wem/structures/FieldGroups.testcase [hunk 1]
```diff
@@ -89,12 +89,57 @@ definition {
...
+		DataEngine.addField(
+			fieldFieldLabel = "Text",
+			fieldName = "Text");
+
+		DataEngine.addFieldNested(
+			fieldFieldLabel = "Numeric",
+			fieldName = "Numeric",
+			targetFieldLabel = "Text");
+
+		DataEngine.addFieldNested(
```
Comment (Tim-Cao) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/tests/wem/structures/FieldGroups.testcase:117]: Target Field should be Text

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/wem/structures/FieldGroups.testcase [hunk 2]
```diff
@@ -89,12 +89,57 @@ definition {
...
+
+		LexiconEntry.gotoEntryMenuItem(
+			menuItem = "Edit",
+			rowEntry = "WC Structure Name");
+
+		AssertTextEquals(
+			locator1 = "DataEngineBuilder#NESTED_DUPLICATED_FORM_FIELDS_GROUP",
+			value1 = "Copy of Fields Group");
+
+		WebContentStructures.saveCP(structureName = "${structureName}");
```
Comment (Tim-Cao) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/tests/wem/structures/FieldGroups.testcase:142]: Unnecessary step

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/path/DataEngineBuilder.path [hunk 3]
```diff
@@ -13,16 +13,45 @@
 	<td>xpath=(//div[contains(@class, 'fieldset')]//label[text()='Fields Group'])[2]</td>
 	<td></td>
 </tr>
+
```
Comment (Tim-Cao) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/tests/path/DataEngineBuilder.path:16]: Use existing locator

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/path/DataEngineBuilder.path [hunk 4]
```diff
@@ -13,16 +13,45 @@
...
+<tr>
+	<td>FORM_FIELDS_GROUP_FIRST</td>
```
Comment (Tim-Cao) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/tests/path/DataEngineBuilder.path:18]: Use existing locator

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/path/DataEngineBuilder.path [hunk 5]
```diff
@@ -13,16 +13,45 @@
...
 </tr>
+
 <tr>
 	<td>DUPLICATED_FORM_FIELDS_GROUP_LABEL</td>
 	<td>//input[(@class='form-control ddm-field-text') and (@placeholder='Enter a field label.')]</td>
 	<td></td>
 </tr>
+
+<tr>
+	<td>NESTED_DUPLICATED_FORM_FIELDS_GROUP</td>
```
Comment (Tim-Cao) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/tests/path/DataEngineBuilder.path:36]: Use existing locator

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/path/DataEngineBuilder.path [hunk 6]
```diff
@@ -13,16 +13,45 @@
...
+	<td>//div[@class='ddm-field-types-fieldset__nested']//label[text()='Copy of Fields Group']</td>
+	<td></td>
+</tr>
+
+<tr>
```
Comment (Tim-Cao) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/tests/path/DataEngineBuilder.path:41]: Use existing locator

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/path/DataEngineBuilder.path [hunk 7]
```diff
@@ -13,16 +13,45 @@
...
+	<td>NESTED_NUMERIC_STRUCTURE</td>
+	<td>//label[text()='Numeric']</td>
+	<td></td>
+</tr>
+
+<tr>
```
Comment (Tim-Cao) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/tests/path/DataEngineBuilder.path:47]: Use existing locator

**Comment by Tim-Cao** [comment]
See https://github.com/liferay-echo/liferay-portal/pull/3628
View total diff https://github.com/liferay-echo/liferay-portal/pull/3628/commits/6fb7732a3f5dc28b37fdce3896d2e4324f3c79cf

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**